conn.go

```diff
@@ -118,12 +118,12 @@ func (c *Conn) Streams() []*Stream {
 // Close closes this connection
 func (c *Conn) Close() error {
 	c.closeLock.Lock()
-	defer c.closeLock.Unlock()
-
 	if c.closed {
+		c.closeLock.Unlock()
```

Comment:
The issue is that most callers of `Close` don't really care to wait until it completes, they just perform some routine cleanup as part of some other method. Asking them all to wait until its 'done' isnt what we want (either that or we need a `BeginClose` method, or some way of signifying we dont care to wait)